src/main/java/org/elasticsearch/index/mapper/DocumentMapper.java

```diff
@@ -558,15 +533,20 @@ public ParsedDocument parse(SourceToParse source, @Nullable ParseListener listen
     /**
      * Returns the best nested {@link ObjectMapper} instances that is in the scope of the specified nested docId.
      */
-    public ObjectMapper findNestedObjectMapper(int nestedDocId, BitsetFilterCache cache, LeafReaderContext context) throws IOException {
+    public ObjectMapper findNestedObjectMapper(int nestedDocId, SearchContext sc, LeafReaderContext context) throws IOException {
         ObjectMapper nestedObjectMapper = null;
         for (ObjectMapper objectMapper : objectMappers().values()) {
             if (!objectMapper.nested().isNested()) {
                 continue;
             }
 
-            BitDocIdSet nestedTypeBitSet = cache.getBitDocIdSetFilter(objectMapper.nestedTypeFilter()).getDocIdSet(context);
-            if (nestedTypeBitSet != null && nestedTypeBitSet.bits().get(nestedDocId)) {
+            Filter filter = sc.filterCache().cache(objectMapper.nestedTypeFilter(), null, sc.queryParserService().autoFilterCachePolicy());
+            if (filter == null) {
+                continue;
+            }
+            DocIdSet nestedTypeSet = filter.getDocIdSet(context, context.reader().getLiveDocs());
```

Comment:
it just felt safe to do this... but I don't we need to, because nested docs can't be modified on their own. Their life cycle depends on the parent doc.